packages/node/src/indexer/worker/worker.unfinalizedBlocks.service.ts

```diff
@@ -22,11 +22,14 @@ export class WorkerUnfinalizedBlocksService
   }
 
   async processUnfinalizedBlocks(
-    block: BlockContent,
+    block: IBlock<BlockContent>,
   ): Promise<number | undefined> {
-    return this.host.unfinalizedBlocksProcess(
-      substrateHeaderToHeader(block.block.block.header),
-    );
+    const header = block.getHeader();
+    return this.host.unfinalizedBlocksProcess({
+      blockHeight: header.height,
+      blockHash: header.hash,
+      parentHash: header.parentHash,
+    });
```

Comment:
We should update unfinalized blocks so the types match and its called like this

```suggestion
    return this.host.unfinalizedBlocksProcess(header);
```